### DISCUSSION TITLE
hugo server testing efficiency - ideas?

### DISCUSSION THREAD

**Discussion by gerbrent** [JupiterBroadcasting/jupiterbroadcasting.com#52] [Q&A]
I am attempting to decrease the time it takes hugo server running locally on my workstation to rebuild the site on `config.toml` changes.

Currently building the site from scratch locally (for me) takes 5+ minutes, which is not so comfortable/efficient/tolerable in quick-iteration testing. 

Note: changing other files leads to very quick updates, so no issue there.

Any thoughts on how to increase efficiency only during testing?

Some ideas:
* ignore most of the show back catalogues, which correlates to the long hugo build times. Perhaps via some form of git ignoring the files during fetch/push.

Any ideas?

**Comment by StefanS-O** [comment]
You could make a branch and remove most of the episodes from the content folder.
It's 20K pages right now, that is quite a lot. On my Macbook Air M1 startup time is around 20 seconds, so i don't really mind.

**Reply by gerbrent** [reply]
So, you're really saying I should beg to have Chris send me the new HP Dev One ; )

**Comment by elreydetoda** [comment]
I thought I was the only one who it was taking a long time for :sweat_smile: 

So, [the PR I did](https://github.com/JupiterBroadcasting/jupiterbroadcasting.com/pull/130), with the RSS feed no longer getting generated, should help out a lot (at least it did on my machine (image in the PR shows the time reduction)).

Also, if we do the redo of the `data/` folder like [I'm recommending here](https://github.com/JupiterBroadcasting/jupiterbroadcasting.com/issues/110), that should help quite a bit as well because I believe hugo can more efficiently load pages than the raw json files we're doing now.

With each raw JSON file the golang JSON parser (while still crazy fast for how much we've got :sweat_smile: ) has to parse each json file and craft it into a long dictionary of dictionaries. While pages is essentially like an object (at least from my understanding, whenever I've used hugo's `print` function on a page) and the pages construct has been really optimized for hugo (since that's normally what it's parsing a majority of the time).



( saw this discussion after #106 mentioning discussions :upside_down_face:  )
